optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -186,20 +186,109 @@
     // Group by
     // ************************************************************************
 
-    <GroupKey_, ResultContainer_, Result_> UniConstraintStream<Result_> groupBy(
+    /**
+     * Runs all items in the stream through a given @{@link UniConstraintCollector} and converts them into a new
+     * {@link UniConstraintStream} which only has one item in it, the result of applying
+     * {@link UniConstraintCollector#finisher()}.
```

Comment:
the result of applying {@link UniConstraintCollector#finisher()}
-> 
the result of applying {@link UniConstraintCollector}

The finisher is a impl detail for 99% of our users (even though it's in the public api...)